### ISSUE TITLE
Add an option to specify the name of the constraint (primary key, unique key, and foreign key)

### ISSUE THREAD

**Issue by mjahammel** [Snowflake-Labs/dbt_constraints#34]
It would be nice to, when desired, override the generated name of the constraint.

**Comment by sfc-gh-dflippo** [comment]
Are you thinking that you would like to set a constraint name for a specific test or are you hoping to specify a different naming convention?

**Comment by mjahammel** [comment]
Something like so:
```
- name: dim_duplicate_orders
    description: "Test that we do not try to create PK/UK on failed tests"
    columns:
      - name: o_orderkey
        description: "The primary key for this table"
      - name: o_orderkey_seq
        description: "duplicate seq column to test UK"
    tests:
      - dbt_constraints.primary_key:
          column_name: o_orderkey
          constraint_name: ddo_pk
          config:
            severity: warn
      - dbt_constraints.unique_key:
          column_name: o_orderkey_seq
          constraint_name: ddo_uk
        config:
            severity: warn
```
The value for 'constraint_name' would be used to name the constraint on the table rather than the generated name.  Not specifying a constraint name would use the current behaviour.  Sorry, I should have put this in the original message.

**Comment by mjahammel** [comment]
I'm still fairly new to dbt/jinja2, but I went ahead and tried implementing it myself.  Attached is a copy of my clone of the git repository with the changes in a branch named '34-optional-constraint-name'.  I made a slight change from my example above.  Instead of using a generic 'constraint_name' parameter for each of the key types, I used distinct parameters: 'pk_name', 'uk_name' and 'fk_name' .

[dbt_constraints-34-optional-constraint-name.zip](https://github.com/Snowflake-Labs/dbt_constraints/files/10043454/dbt_constraints-34-optional-constraint-name.zip)
